Telemetry/TelemetryStorage.swift

```diff
@@ -11,37 +11,51 @@ import Foundation
 class TelemetryStorageSequence : Sequence, IteratorProtocol {
     typealias Element = TelemetryPing
 
-    private let directoryEnumerator: FileManager.DirectoryEnumerator?
+    private let files: [URL]
+    
+    private var index = -1
 
     private var currentPing: TelemetryPing?
     private var currentPingFile: URL?
 
-    init(directoryEnumerator: FileManager.DirectoryEnumerator?) {
-        self.directoryEnumerator = directoryEnumerator
+    init(files: [URL]) {
+        self.files = files
     }
 
     func next() -> TelemetryPing? {
-        guard let directoryEnumerator = self.directoryEnumerator else {
-            return nil
-        }
+        index += 1
+
+        while files.count > index {
+            let url = files[index]
+            
+            var isDirectory: ObjCBool = false
+
+            guard FileManager.default.fileExists(atPath: url.path, isDirectory: &isDirectory), !isDirectory.boolValue else {
+                index += 1
+                continue
+            }
+
+            currentPingFile = url
```

Comment:
is this var needed now that there is a current index? 
Could it be a getter to return files[index]